std/file.d

```diff
@@ -1782,6 +1784,50 @@ if (isInputRange!R && !isInfinite!R && isSomeChar!(ElementEncodingType!R))
     assert(target.timeLastModified(SysTime.min) >= source.timeLastModified);
 }
 
+version(StdDdoc)
+{
+    /++
+     $(BLUE This function is Posix-Only.)
+
+     Returns the time that the given file was last modified.
+     Params:
+        statbuf = stat_t retrieved from file.
+     +/
+    SysTime timeLastModified(ref stat_t statbuf) pure nothrow;
```

Comment:
Why are any of these `ref`? That's just going to be really annoying any time someone has an rvalue or if the argument is `const`. The function is not modifying the argument, so it really shouldn't be `ref`. If you're really that worried about copying the `stat_t`, then just templatize these functions and make them take `auto ref` - e.g.

    SysTime timeLastModified()(auto ref stat_t statbuf);

or

    SysTime timeLastModified(T)(auto ref T statbuf)
    if (is(Unqual!T == stat_t))
    {...}

Reply:
I think the compiler has been automatically templatizing `auto ref` functions, so `()(` is not necessary.

Reply:
If so, that's a new thing, but I just tried it locally, and the compiler still gives an error if you use `auto ref` and the function isn't templated.

Reply:
I should probably point out though that if these are templated instead of just accepting `stat_t` without `ref`, then their bodies in the `version(StdDoc)` section will probably need `assert(0);` in them to make the compiler shut up about the lack of function body for a templated function.